### DISCUSSION TITLE
publish to Confluence with params

### DISCUSSION THREAD

**Discussion by hutch3232** [quarto-dev/quarto-cli#9830] [Q&A] [answered]
### Description

Firstly, I wanted to mention that I found this related discussion: https://github.com/quarto-dev/quarto-cli/discussions/5310

In that discussion, it sounded like the recommended approach (until `publish` inherits from `render`) is to do a two-stage approach: first rendering the document with parameters supplied and then to publish using the `--no-render` option.

Unfortunately, when I tried to do that I get:
`ERROR: Confluence requires rendering before publish`

My code:
```
quarto render confl.qmd -P myvar:"test"
quarto publish confluence --no-render confl.qmd
```

Am I approaching this wrong? Is it possible to pass render-time parameters to a Confluence-bound .qmd?

```
quarto --version
1.4.550
Ubuntu 20.04
```

**Comment by cderv** [comment, marked as answer]
Thanks for sharing. This is indeed a limitation of the workflow. `quarto publish` does not take render parameters, and to publish to confluence, this is using a special internal format used when doing `quarto publish`. 

I'll open an issue to track the idea for improvement

**Reply by mcanouil** [reply]
- https://github.com/quarto-dev/quarto-cli/issues/9866

**Comment by SimonCoulombe** [comment]
What would be a good workaround until this is fixed?  I'm considering just rendering a bunch of parameterised reports to .html with different parameters then uploading these html as attachment to confluence pages.
I'm using   conflr::confl_post_attachment() and  conflr::confl_update_attachment_data() to do that
